### PR TITLE
Orders: improving updateWithPositionHint to updateAmounts in case no move required

### PR DESCRIPTION
**Pull request by amitkot**
Also some test ordering and renaming.

### CODE REVIEW

**Comment by amitkot** [comment]
@ilanDoron please review.

File: contracts/permissionless/Orders.sol [hunk 1]
```diff
@@ -138,6 +126,46 @@ contract Orders is Withdrawable, Utils2, OrdersInterface {
...
+    function updateAmounts(
+        uint32 orderId,
+        uint128 srcAmount,
+        uint128 dstAmount
+    )
+        public
+        onlyAdmin
+        returns (bool)
+    {
+        require(orderId != HEAD_ID && orderId != TAIL_ID);
```
Comment (ilanDoron) [inline contracts/permissionless/Orders.sol]: code for this function is a copy of the code in isRightPosition

if isRightPosition will have another parameter (nextId) u will be able to use it for this.

File: contracts/permissionless/Orders.sol [hunk 2]
```diff
@@ -116,16 +114,6 @@ contract Orders is Withdrawable, Utils2, OrdersInterface {
...
-        bool hintIsRightPositionAfterUpdate = false;
-        if (hintIsCurrentPosition && hintIsRightPositionAfterUpdate) {
-            // Order is in the right position, update amounts
-            orders[orderId].srcAmount = srcAmount;
-            orders[orderId].dstAmount = dstAmount;
-            AmountUpdateOnly();
-            return true;
-        }
-
         if (isRightPosition(srcAmount, dstAmount, prevId)) {
```
Comment (ilanDoron) [inline contracts/permissionless/Orders.sol]: if the order doesn't change position here the test might fail while still the correct position.
I know in our design it shouldn't happen. but still. its a flow in the code.
I can build a test that will prove it....

Comment (ilanDoron) [reply contracts/permissionless/Orders.sol]: I mean. the correct test will make this update call fail...

File: contracts/permissionless/Orders.sol (hunk 1 above)
Comment (amitkot) [reply contracts/permissionless/Orders.sol]: updateAmounts merged into updateWithPositionHint, using a single validation function as we discussed f2f.

File: contracts/permissionless/Orders.sol (hunk 2 above)
Comment (amitkot) [reply contracts/permissionless/Orders.sol]: Agree, added test and fixed the issue.

**Comment by amitkot** [comment]
@ilanDoron merged updateAmounts to udpateWithHint. Please review and merge.

File: contracts/permissionless/Orders.sol [hunk 3]
```diff
@@ -213,38 +226,41 @@ contract Orders is Withdrawable, Utils2, OrdersInterface {
...
+        uint32 prevId,
+        uint32 nextId
     )
         private
         view
         returns (bool)
     {
-        // Make sure prev is not the tail.
-        if (prevId == TAIL_ID) return false;
+        if (prevId == TAIL_ID || nextId == HEAD_ID) return false;
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contracts/permissionless/Orders.sol]: sorry. its testing another thing. my error...

**Pull request merged**